### PR TITLE
FE: Revert #25061; disable tables during their initial sync on data model page

### PR DESCRIPTION
**Pull request by calherries**
Epic: https://github.com/metabase/metabase/issues/31082
[Additional context](https://metaboat.slack.com/archives/C057T1QTB3L/p1686909790028459)

We should revert the changes to the `MetadataTableList` and `MetadataTable` components in https://github.com/metabase/metabase/pull/25061 because (a) it's broken now and (b) it is unnecessary because the problem it attempted to solve is fixed by [this PR](https://github.com/metabase/metabase/pull/25945).

### CODE REVIEW

**Comment by WiNloSt** [comment]
@calherries Can you fix the link for the PR that's fixing the issue? It's still linking to https://github.com/metabase/metabase/pull/25061 which is the PR you said you're reverting.

**Comment by calherries** [comment]
@WiNloSt thanks, fixed

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.tsx [hunk 1]
```diff
@@ -260,18 +263,20 @@ const TableRow = ({
       <a
         className={cx(
           "AdminList-item flex align-center no-decoration text-wrap justify-between",
-          { selected: isSelected },
+          { selected: isSelected, disabled: !isSyncCompleted(table) },
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.tsx:266]: This uses a very old CSS declaration

https://github.com/metabase/metabase/blob/86c4848035b23027ff11ef84b6fec1a29a7f6c53/frontend/src/metabase/css/admin.css#L119-L122

could you maybe migrate this to styled component and pass `disabled` prop directly? The pros are that it will also automatically get `disabled` prop on the dom node, and you could assert in Cypress using `should("be.disabled")` instead.

Just to be clear you can use this style in the styled component file (there's none, so you need to create a new file)

```ts
   color: ${color("text-medium")}; 
   pointer-events: none; 
   opacity: 0.4;
```

These are the styles I got from inspecting the element. You should double check.

**Review by WiNloSt — COMMENTED** [review]
Looks good. Could you help address the use of old CSS class name that's discouraged and it should be good to go.

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.tsx (hunk 1 above)
Comment (calherries) [reply frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.tsx]: Okay, I think tried to create a new styled component in [Create AdminListItem styled component](https://github.com/metabase/metabase/pull/31661/commits/e9708e0ea589ad165d91c1197a848745a53038c3). 

One issue is I don't know how to prevent duplication when converting the `selected` class into a prop.

This was in the css before:
```
.AdminList-item.selected,
.AdminList-item:hover {
  background-color: white;
  border-color: var(--color-border);
  margin-left: -0.5em;
  margin-right: -0.5em;
  padding-left: 1.5em;
  padding-right: 1.5em;
  box-shadow: 0 1px 2px var(--color-shadow);
}
```

And now I have duplicated the css in the `AdminListItem` styled component:
```
  ${({ selected }) =>
    selected &&
    css`
      background-color: white;
      border-color: var(--color-border);
      margin-left: -0.5em;
      margin-right: -0.5em;
      padding-left: 1.5em;
      padding-right: 1.5em;
      box-shadow: 0 1px 2px var(--color-shadow);
    `};
  &:hover {
    background-color: white;
    border-color: color("border");
    margin-left: -0.5em;
    margin-right: -0.5em;
    padding-left: 1.5em;
    padding-right: 1.5em;
    box-shadow: 0 1px 2px ${color("shadow")};
  }
```

I could remove the duplication by extracting the css props out to a varaible and name it something like `selectedOrHoverStyle`, but I'd like to avoid having to name it and extract it out. Is there another way?

Comment (WiNloSt) [reply frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.tsx]: At first, I was thinking about only migrating `disabled`–which is what you've added–but the entire component is even better.

Unfortunately there's no way to go around this as far as I can tell, previously we can use the same block since we use 2 CSS selectors, but since we're passing props directly we can't really combine dynamic style + CSS selector, in this case, `selected` + `:hover`.

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  word-wrap: anywhere;
+  justify-content: space-between;
+  color: ${({ selected }) =>
+    selected ? color("brand") : color("text-medium")};
+
+  ${({ selected }) =>
+    selected &&
+    css`
+      background-color: white;
+      border-color: var(--color-border);
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx:29]: nit:
```suggestion
      border-color: ${color("border")};
```

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      margin-left: -0.5em;
+      margin-right: -0.5em;
+      padding-left: 1.5em;
+      padding-right: 1.5em;
+      box-shadow: 0 1px 2px var(--color-shadow);
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx:34]: nit:
```suggestion
      box-shadow: 0 1px 2px ${color("shadow")};
```

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    `};
+
+  &:hover {
+    background-color: white;
+    border-color: color("border");
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx:39]: `color` should be a JS function
```suggestion
    border-color: ${color("border")};
```

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+interface AdminListItemProps {
+  selected?: boolean;
+  disabled?: boolean;
+}
+
+export const AdminListItem = styled.a<AdminListItemProps>`
+  padding: 0.75em 1em 0.75em 1em;
+  border: var(--border-size) var(--border-style) transparent;
+  border-radius: var(--default-border-radius);
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataTableList/MetadataTableList.styled.tsx:13]: Safe to say this can be hard-coded as well
```suggestion
  border-radius: 8px;
```

I couldn't find any usage of these CSS custom properties except `var(--default-font-family)` which we can assume we don't use CSS custom properties (`--custom-properties` thing) in styled-components anymore.

**Review by WiNloSt — COMMENTED** [review]
It's a great start on migrating to styled components. I've commented on some typo + improvements then it'll be good to go.

Thank you for addressing the feedback. You doing great.

**Review by WiNloSt — APPROVED** [review]
Awesome! Thanks for addressing the feedback. I haven't tested the actual style though.

**Comment by calherries** [comment]
@WiNloSt I restored the original AdminList-item CSS and removed the CSS from the styled component except for the `disabled` prop. See [Restore AdminList-item css](https://github.com/metabase/metabase/pull/31661/commits/52ee72cf39a86578433260c4ebdedbc26abc9efe). The reason is `AdminList-item` is used in other places and I didn't want to risk any changes there.

**Comment by calherries** [comment]
I also changed the test to test for `pointer-events: none` since the `.toBeDisabled()` check doesn't seem to work for anchor elements, only buttons.

File: frontend/src/metabase/admin/datamodel/metadata/components/MetadataEditor/MetadataEditor.unit.spec.tsx [hunk 6]
```diff
@@ -236,6 +245,18 @@ describe("MetadataEditor", () => {
...
     });
 
+    it("should display tables with initial_sync_status='incomplete' as disabled", async () => {
+      await setup({ databases: [SAMPLE_DB_WITH_INITIAL_SYNC_INCOMPLETE] });
+      expect(
+        screen.getByText(SAMPLE_DB_WITH_INITIAL_SYNC_INCOMPLETE.name),
+      ).toBeInTheDocument();
+      expect(await screen.findByText("1 Queryable Table")).toBeInTheDocument();
+      const computedStyles = window.getComputedStyle(
+        screen.getByText(ORDERS_TABLE_INITIAL_SYNC_INCOMPLETE.display_name),
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/metadata/components/MetadataEditor/MetadataEditor.unit.spec.tsx:255]: What about testing that you can't click this table by clicking the table and asserting that the right side isn't rendered e.g. you can query that `Select any table to see its schema and add or edit metadata.` is in the screen
image


This would match user behavior more than just asserting style on the DOM nodes directly which we should avoid.

**Comment by calherries** [comment]
I rebased on https://github.com/metabase/metabase/pull/31773 to get CI to pass. Will merge after the other PR is merged
